### DISCUSSION TITLE
Looking for F' CMake function to process framework directives.

### DISCUSSION THREAD

**Discussion by Austen-G** [nasa/fprime#1770] [Q&A] [answered]
Hello everyone! I am currently working on writing unit tests for components under test that is intended to run on a Xilinx MicroBlaze processor. We have F' cross-compiling for this processor, but are having trouble with unit testing these components using the Linux gcc compiler. 

The compiler for our MicroBlaze processor adds this directive: `#define __MICROBLAZE__` behind the scenes which enables the BSP code to select functions that are specific to the processor. This is fine when compiling for hardware, but the Linux gcc compiler used for the unit tests has no idea of its existence. 

We have already tried the `add_definitions` and `add_compile_definition` commands to no avail, and are wondering if there is a custom CMake function from F' that accomplishes this. We would like to avoid using another header file and #ifdef to accomplish this if possible.

**Comment by kevin-f-ortega** [comment, marked as answer]
Using `add_definitions` or `add_compile_definition` should have added it. You can build your component using the `--verbose` flag. That will show you the command used to compile your code; so, you should see your definition there. Just make sure you slightly modify a cpp file in order to trick CMake into re-building your component.

**Reply by Austen-G** [reply]
Oops it looked like there was some confusion regarding the syntax on our end, using: `add_compile_definition()` without -D worked. Thank you!
